File: csharp/HoneydewTestProject/NotAmbiguous.cs
using HoneydewTestProject.A;

namespace HoneydewTestProject
{
    public class NotAmbiguous
    {
        private D1 d1Field;

        public D1 D1Prop { get; set; }

        public void UsingD1AsLocalVariable()
        {
            D1 localVariable = new D1();
        }

        public void UsingD1AsLocalVariable2()
        {
            D1 localVariable = new D1();
        }

        public void UsingD1AsParameter(D1 param)
        {
            var x = param.X;
        }
    }
}

File: vb/HoneydewTestProjectVB/NotAmbiguous.vb
Imports HoneydewTestProjectVB.A

Namespace HoneydewTestProjectVB
    Public Class NotAmbiguous
        Private d1Field As D1
        Public Property D1Prop As D1

        Public Sub UsingD1AsLocalVariable()
            Dim localVariable As D1 = New D1()
        End Sub

        Public Sub UsingD1AsLocalVariable2()
            Dim localVariable As D1 = New D1()
        End Sub

        Public Sub UsingD1AsParameter(ByVal param As D1)
            Dim x = param.X
        End Sub
    End Class
End Namespace
